tempesta_fw/sock_srv.c

```diff
@@ -889,11 +1311,23 @@ tfw_cfgop_in_server(TfwCfgSpec *cs, TfwCfgEntry *ce)
 static int
 tfw_cfgop_out_server(TfwCfgSpec *cs, TfwCfgEntry *ce)
 {
-	if (tfw_cfgop_server(cs, ce, &tfw_cfg_out_slst))
+	if (!tfw_cfg_sg_def) {
+		TFW_ERR_NL("'default' group is declared implicitly after "
+			   "explicit declaration.\n");
 		return -EINVAL;
-	tfw_cfg_out_slstsz++;
+	}
```

Comment:
If in config file a server outside any group is defined at first and then default group is defined explicitly - is this an acceptable case?

Reply:
No, it's not acceptable. Once a server is defined outside of any group, `default` group will be registered as available after reconfiguration. If the explicit `default` group is defined after that, `tfw_cfgop_begin_srv_group()` will throw a `Group '...' already exists` error.

Reply:
Yes - got it. Thanks.